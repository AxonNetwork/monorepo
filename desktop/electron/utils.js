const path = require('path')

module.exports.getAppPath = function() {
    try {
        return require('electron').app.getAppPath()
    } catch (err) {
        return require('electron').remote.app.getAppPath()
    }
}

module.exports.getEnv = function() {
    const appPath = module.exports.getAppPath()
    const binariesPath = path.join(appPath, '../desktop/build-resources/binaries')
    const env = Object.assign({}, process.env, { CONSCIENCE_APP_PATH: appPath, CONSCIENCE_BINARIES_PATH: binariesPath })
    env.PATH = binariesPath + ':' + env.PATH + ':/usr/local/bin:/usr/bin:/usr/sbin:/sbin'
    return env
}
