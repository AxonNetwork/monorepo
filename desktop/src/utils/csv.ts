export function parseCSV(contents: string){
	const lines = contents.split("\n")
	const data = lines.map(line => line.split(","))
	return data
}