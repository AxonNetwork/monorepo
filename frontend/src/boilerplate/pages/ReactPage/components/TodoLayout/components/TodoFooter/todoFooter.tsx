import * as React from 'react';
import { InjectedI18nProps, InjectedTranslateProps, translate } from 'react-i18next';

import TodoFilter from './components/TodoFilter';
import i18ns from './i18n';

interface ITodoFooterProps extends InjectedI18nProps, InjectedTranslateProps { }

export class TodoFooter extends React.Component<ITodoFooterProps> {
  constructor(props: ITodoFooterProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'TodoFooter', i18ns[key]);
    });
  }

  public render() {
    const { t } = this.props;
    return (
      <div className='todo-footer'>
        <TodoFilter filter='SHOW_ALL'>
          {t('all')}
        </TodoFilter>
        <TodoFilter filter='SHOW_ACTIVE'>
          {t('active')}
        </TodoFilter>
        <TodoFilter filter='SHOW_COMPLETED'>
          {t('completed')}
        </TodoFilter>
      </div>
    );
  }
}

export default translate('TodoFooter')(TodoFooter);
