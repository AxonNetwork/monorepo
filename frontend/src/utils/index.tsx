export const isProduction = process.env.NODE_ENV === 'production'

const exec = require('child_process').exec

export async function execCmd(cmd: string, cwd: string) {
    return new Promise((resolve, reject) => {
        exec(cmd, { encoding: 'utf8', cwd }, (error: Error|null, stdout: string, stderr: string) => {
            if (error) {
                reject(error)
            } else {
                resolve({ stdout, stderr })
            }
        })
    })
}
