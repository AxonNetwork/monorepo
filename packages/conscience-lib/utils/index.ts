export { default as autobind } from './autobind'
export { default as strToColor } from './strToColor'
export { default as schemes } from './codeColorSchemes'
export { default as getLanguage } from './getLanguage'
export { default as checkVisible } from './checkVisible'
export { default as nonCacheImg } from './nonCacheImg'
export { repoPageToString, stringToRepoPage } from './parseRepoPage'
export { orgPageToString, stringToOrgPage } from './parseOrgPage'
export { getConscienceURI } from './uri'
export { removeEmail, extractEmail } from './email'
export { parseCSV } from './csv'
export { isTextFile } from './fileTypes'
export { getLastVerifiedEvent, getLastVerifiedEventCommit, getLastVerifiedEventFile, getFirstVerifiedEvent } from './timeline'
