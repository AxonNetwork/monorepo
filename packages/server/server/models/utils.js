import { dynamo } from '../config/aws'

export function makeID() {
    let text = ''
    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'

    for (let i = 0; i < 10; i++) {
        text += possible.charAt(Math.floor(Math.random() * possible.length))
    }

    return text
}

export async function getAll(queryParams) {
    let items = []
    let ExclusiveStartKey

    do {
        const resp = await dynamo.queryAsync({
            ...queryParams,
            ExclusiveStartKey,
        })

        if (resp.Items) {
            items = items.concat(resp.Items)
        }

        ExclusiveStartKey = resp.LastEvaluatedKey
    } while (ExclusiveStartKey)

    return items
}
