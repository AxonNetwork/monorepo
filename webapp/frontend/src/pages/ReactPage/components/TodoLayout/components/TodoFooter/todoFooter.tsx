import React from 'react';
import { withNamespaces, WithNamespaces as Ii18nProps } from 'react-i18next';

import TodoFilter from './components/TodoFilter';
import i18ns from './i18n';

export class TodoFooter extends React.Component<Ii18nProps> {
  constructor(props: Ii18nProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'TodoFooter', i18ns[key]);
    });
  }

  public render() {
    const { t } = this.props;
    return (
      <div className='todo-footer'>
        <TodoFilter filter='SHOW_ALL'>
          {t('all')}
        </TodoFilter>
        <TodoFilter filter='SHOW_ACTIVE'>
          {t('active')}
        </TodoFilter>
        <TodoFilter filter='SHOW_COMPLETED'>
          {t('completed')}
        </TodoFilter>
      </div>
    );
  }
}

export default withNamespaces('TodoFooter')(TodoFooter);
