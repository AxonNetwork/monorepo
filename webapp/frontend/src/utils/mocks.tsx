// Mock dispatch
export const dispatchMock = () => { return; };

// Mock translation
export const tMock = (k: string | string[]) => k;

// Mock i18n
export const i18nMock: any = {
  addResourceBundle: () => { return; },
};
