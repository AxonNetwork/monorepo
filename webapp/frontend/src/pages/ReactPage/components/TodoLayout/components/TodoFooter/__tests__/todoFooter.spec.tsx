import React from 'react';
import TestRenderer from 'react-test-renderer';

import { i18nMock, tMock } from 'utils/mocks';
import { TodoFooter } from '../todoFooter';

jest.mock('../components/TodoFilter', () => 'TodoFilter');

describe('TodoFooter', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <TodoFooter t={tMock} tReady={true} i18n={i18nMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
