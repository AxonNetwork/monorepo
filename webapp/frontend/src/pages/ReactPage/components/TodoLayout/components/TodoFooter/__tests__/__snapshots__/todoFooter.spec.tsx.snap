// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFooter should renders correctly 1`] = `
<div
  className="todo-footer"
>
  <TodoFilter
    filter="SHOW_ALL"
  >
    all
  </TodoFilter>
  <TodoFilter
    filter="SHOW_ACTIVE"
  >
    active
  </TodoFilter>
  <TodoFilter
    filter="SHOW_COMPLETED"
  >
    completed
  </TodoFilter>
</div>
`;
